.github/workflows/build-on-pr.yml

```diff
@@ -1,20 +1,29 @@
 name: Build Docker Image
 
 on:
-  push:
-    branches: 
-      - '*'
-      - '!main'
   pull_request:
-    branches: [ '*' ]
 
 jobs:
   build:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
     steps:
     - uses: actions/checkout@v4
-
-    - name: Build Docker image
-      run: |
-        cd kratos-admin-ui
-        docker build .
\ No newline at end of file
+    - uses: docker/setup-buildx-action@v2
+    - uses: docker/login-action@v2
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - id: meta
```

Comment:
What's the benefit of using this actions instead of running the command directly?